WooCommerce/WooCommerceUITests/Screens/MyStore/MyStoreScreen.swift

```diff
@@ -1,39 +1,39 @@
-import Foundation
+import ScreenObject
 import XCTest
 
-final class MyStoreScreen: BaseScreen {
-
-    struct ElementStringIDs {
-        static let topBannerCloseButton = "top-banner-view-dismiss-button"
-        static let settingsButton = "dashboard-settings-button"
-    }
+final class MyStoreScreen: ScreenObject {
 
     let tabBar = TabNavComponent()
     let periodStatsTable = PeriodStatsTable()
-    private let settingsButton: XCUIElement
-    private let topBannerCloseButton: XCUIElement
 
-    static var isVisible: Bool {
-        let settingsButton = XCUIApplication().buttons[ElementStringIDs.settingsButton]
-        return settingsButton.exists && settingsButton.isHittable
+    private let settingsButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons["dashboard-settings-button"]
+    }
+
+    private let topBannerCloseButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons["top-banner-view-dismiss-button"]
     }
 
-    init() {
-        settingsButton = XCUIApplication().buttons[ElementStringIDs.settingsButton]
-        topBannerCloseButton = XCUIApplication().buttons[ElementStringIDs.topBannerCloseButton]
+    private var settingsButton: XCUIElement { settingsButtonGetter(app) }
+    private var topBannerCloseButton: XCUIElement { topBannerCloseButtonGetter(app) }
```

Comment:
We use the getter instance properties to [_compute_ ](https://docs.swift.org/swift-book/LanguageGuide/Properties.html#ID259) the element at runtime so it's always up to date.